fix: symeig depreciated, switched to torch.linalg.eigh.

hlzl opened the pull request:
`symeig` no longer works with `torch>=2.0.0`. `torch.linalg.eigh` is its replacement.

f-dangel committed:
[DOC] Update citation: arXiv → JMLR

hlzl committed:
fix: symeig depreciated, switched to torch.linalg.eigh.

f-dangel commented:
Hi, thanks for pointing this out. The package uses `symeig` in a couple of places because it runs on GPU, in contrast to `torch.linalg.eigh`. Since this package is an add-on to BackPACK which now officially supports `torch>=2`, I think it's a good idea to remove the deprecated `symeig` calls from this library as well (and also to deprecate `python3.7`). Could you pull-request into the [`torch-2`](https://github.com/f-dangel/vivit/tree/torch-2) branch and use the same import style (`from torch.linalg import eigh`)? I will take care of deprecating `python3.7`.

hlzl committed:
fix: Changed remaining symeig to eigh.

hlzl committed:
chore: Updated import style.

hlzl committed:
Merge branch 'torch-2' into symeig-to-eigh

hlzl commented:
Hi Felix,
I updated the remaining cases where `torch`'s `symeig` was used and switched the branch.

BTW, to me it looks like `eigh` actually runs on the GPU and is [faster on large matrices](https://discuss.pytorch.org/t/torch-linalg-eigh-is-significantly-slower-on-gpu/140818/8) compared to letting it run on the CPU. It just seems to sync with the CPU at some point, making it slow when using it on many small matrices. Not sure which use case is more important for this package.

Cheers,
Florian

f-dangel commented:
> BTW, to me it looks like eigh actually runs on the GPU and is [faster on large matrices](https://discuss.pytorch.org/t/torch-linalg-eigh-is-significantly-slower-on-gpu/140818/8) compared to letting it run on the CPU. It just seems to sync with the CPU at some point, making it slow when using it on many small matrices. Not sure which use case is more important for this package.

Oh, thanks for clarifying this! Seems like I skimmed its documentation too quickly.

I'll merge this soon and publish a new release that works out of the box with `torch>=2`. Thanks for your contribution!
